pkg/controller/grafana/grafana_reconcile.go

```diff
@@ -485,3 +492,49 @@ func doCheckApplicationMonitoring(r *ReconcileGrafana) (bool, error) {
 	return false, nil
 
 }
+
+func cleanupCSMonitoring(r *ReconcileGrafana, cr *v1alpha1.Grafana) error {
+
+	collectdDep := utils.CollectdDeployment(cr)
+	err := r.client.Delete(r.ctx, collectdDep)
+	if err != nil {
+		log.Info("Failed to clean old Collectd deployment or its already cleaned")
+	}
+
+	kubestateDep := utils.KubestateDeployment(cr)
+	err = r.client.Delete(r.ctx, kubestateDep)
+	if err != nil {
+		log.Info("Failed to clean old kube-state deployment or its already cleaned")
+	}
+
+	nodeExporter := utils.NodeExporterDaemonSet(cr)
+	err = r.client.Delete(r.ctx, nodeExporter)
+	if err != nil {
+		log.Info("Failed to clean old Node exporter daemonset or its already cleaned")
+	}
+
+	promOpr := utils.PrometheusOperatorDeployment(cr)
+	err = r.client.Delete(r.ctx, promOpr)
+	if err != nil {
+		log.Info("Failed to clean old Prometheus operator deployment or its already cleaned")
+	}
+
+	promStatefulset := utils.PrometheusStatefulSet(cr)
+	err = r.client.Delete(r.ctx, promStatefulset)
+	if err != nil {
+		log.Info("Failed to clean old prometheus statefulset or its already cleaned")
+	}
+
+	alertmngrStatefulset := utils.AlertManagerStatefulset(cr)
+	err = r.client.Delete(r.ctx, alertmngrStatefulset)
+	if err != nil {
+		log.Info("Failed to clean old alert-manager statefulset or its already cleaned")
+	}
+
+	mcmCtlDep := utils.McmCtlDeployment(cr)
+	err = r.client.Delete(r.ctx, mcmCtlDep)
+	if err != nil {
+		log.Info("Failed to clean old mcmCtl deployment or its already cleaned")
+	}
+	return err
```

Comment:
please add log entry indicating cleanup suceed (if err not nil)

Reply:
@pgodowski , addressed it in the latest commit